homeassistant/components/generic_thermostat/climate.py

```diff
@@ -266,20 +266,21 @@ def _async_startup(*_):
 
         # Check If we have an old state
         if (old_state := await self.async_get_last_state()) is not None:
-            # If we have no initial temperature, restore
-            if self._target_temp is None:
-                # If we have a previously saved temperature
-                if old_state.attributes.get(ATTR_TEMPERATURE) is None:
-                    if self.ac_mode:
-                        self._target_temp = self.max_temp
-                    else:
-                        self._target_temp = self.min_temp
+
+            # If we have a previously saved temperature restore it
+            if old_state.attributes.get(ATTR_TEMPERATURE) is not None:
+                self._target_temp = float(old_state.attributes[ATTR_TEMPERATURE])
+            elif self._target_temp is None:
+                # If we have no initial temperature, initialize
+                if self.ac_mode:
+                    self._target_temp = self.max_temp
+                else:
+                    self._target_temp = self.min_temp
```

Comment:
Perhaps this would be more concise as:
self._target_temp = self.max_temp if self.ac_mode else self.min_temp

Reply:
Thanks for the reply. I don't know python that much yet. My intention was just to turn around the initial test for the old state and not touch anything else. :)

Reply:
your call. not that important but seems more readable to me.
in any case, i am not an official approver, so it will have to wait until one of them reviews